woocommerce/woocommerce
# Detect cache when launching store

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #48516 .

This PR verifies if the 'coming soon' cache is displayed when launching the store and alerts the user if it is still present.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch.
2. Go to https://your-jn-site/wp-admin/plugin-editor.php
3. Choose `src/Internal/ComingSoon/ComingSoonRequestHandler.php`
4. Return `true` from `should_show_coming_soon` function for testing. 
5. Go to `WooCommerce -> Home` and click on `Launch your store` task
6. Click on `Launch your store` button from the side bar.
7. You should see the copy for the cached site.

![Screen Shot 2024-06-18 at 1 15 24 PM](https://github.com/woocommerce/woocommerce/assets/4723145/fb35307d-4def-433b-af16-af6a3f137922)

Extra:

1. Run the same test without modifying `src/Internal/ComingSoon/ComingSoonRequestHandler.php` on a real site with cache enabled.
2. For the hosting details, see p1718618167417629/1718593364.831769-slack-C01SFMVEYAK



### Changelog entry






-   [] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 


Verify if the coming soon cache is displayed when launching the store and alerts the user if it is still present.

#### Comment 



</details>
